File: frontend/src/types/progress.ts
export interface WeeklyActivity {
    day: string;
    count: number;
}

export interface SkillProgress {
    name: string;
    progress: number;
}

export interface PracticeHistory {
    title: string;
    score: number;
    xpEarned: number;
    createdAt: string;
}

export interface AIInsight {
    strength: string;
    weakness: string;
    recommendation: string;
}

export interface ProgressResponse {
    level: number;

    xp: number;

    nextLevelXp: number;

    completedPractice: number;

    accuracy: number;

    averageScore: number;

    totalCorrect: number;

    totalWrong: number;

    // TAMBAHKAN INI
    streak: number;

    weeklyActivity: WeeklyActivity[];

    skills: SkillProgress[];

    recent: PracticeHistory[];

    insight: AIInsight;
}

File: backend/internal/dto/progress.go
package dto

type WeeklyActivity struct {
	Day   string `json:"day"`
	Count int    `json:"count"`
}

type SkillProgress struct {
	Name     string `json:"name"`
	Progress int    `json:"progress"`
}

type RecentPractice struct {
	Title     string `json:"title"`
	Score     int    `json:"score"`
	XPEarned  int    `json:"xpEarned"`
	CreatedAt string `json:"createdAt"`
}

type AIInsight struct {
	Strength      string `json:"strength"`
	Weakness      string `json:"weakness"`
	Recommendation string `json:"recommendation"`
}

type ProgressResponse struct {
	Level              int                `json:"level"`
	XP                 int                `json:"xp"`
	NextLevelXP        int                `json:"nextLevelXp"`
	CompletedPractice  int                `json:"completedPractice"`
	Accuracy           float64            `json:"accuracy"`
	AverageScore       float64            `json:"averageScore"`
	TotalCorrect       int                `json:"totalCorrect"`
	TotalWrong         int                `json:"totalWrong"`
	Streak             int                `json:"streak"`
	WeeklyActivity     []WeeklyActivity   `json:"weeklyActivity"`
	Skills             []SkillProgress    `json:"skills"`
	Recent             []RecentPractice   `json:"recent"`
	Insight            AIInsight          `json:"insight"`
}